packages/eas-cli/src/credentials/ios/IosCredentialsProvider.ts

```diff
@@ -147,7 +147,7 @@ export default class IosCredentialsProvider implements CredentialsProvider {
   private async fetchRemoteAsync(): Promise<PartialIosCredentials> {
     if (this.options.distribution === DistributionType.INTERNAL) {
       const { app } = this.options;
-      const account = findAccountByName(this.ctx.user.accounts, app.accountName);
+      const account = findAccountByName(this.ctx.user.accounts ?? [], app.accountName);
```

Comment:
This is a side-effect of the `accounts: [Account]` GraphQL typing, which is pretty annoying.

Reply:
This is an unfortunate choice of type in the schema indeed, we should open an issue in www to change the type to `accounts: [Account!]!` (unless there's an issue for that already).

In the meantime would could add a helper like this:
```ts
ensureAccountsArray(accounts: Actor['accounts']): Account[]
```
It would then be used like this:
```ts
findAccountByName(ensureAccountsArray(this.ctx.user.accounts), app.accountName);
```
This would put all the workarounds like `accounts = accounts ?? []` and `filter(account => account)` in one place rather than spreading across the codebase. The idea would be that once the schema has been improved this function could just return the argument unchanged and it would be easy to find all the places that use it and remove it.

Reply:
This is a really good pattern, thanks for sharing. ❤️   I added it to the PR, with a `@todo` notice when the GraphQL types are stricter.